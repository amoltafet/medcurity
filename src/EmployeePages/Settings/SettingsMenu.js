import { Nav, Row, Form, Tab, Col, Container, Button } from 'react-bootstrap';
import { useEffect, useState } from "react";
import axios from 'axios';
import React from 'react';
import './SettingsMenu.css';
import 'bootstrap/dist/css/bootstrap.min.css';

/**
* Creates and displays the Settings menu allows the user to toggle between diffrent settings. 
* @return {GetPage}
*/
const SettingsMenu = () => {
    axios.defaults.withCredentials = true;
    const [session, setSession] = useState([]);
    const [newEmail, setEmail] = useState("");
    const [newUserName, setUsername] = useState("");
    const [saveData, setSaveData] = useState(false);

    useEffect(() => {
        axios.get("http://localhost:3002/users/login").then((response) => {
            console.log('aaahhh', response.data.user)
            setSession(response.data.user[0])
        }).catch(error => console.error(`Error ${error}`));
    }, []);

    useEffect(() => {
        if (saveData === true) {
            axios.put("http://localhost:3002/users/settings", { params: { username: newUserName, id: session.userid } }).then((response) => {
                setSession(response);
            })
            .catch(error => console.log(`Error ${error}`));
        } 
   }, [saveData])

    function SaveUpdatedUserInfo () {
      setSaveData(true);
    }

    console.log(session)

    return (
        <>
            <Tab.Container className="settingsRow" id="left-tabs-example" defaultActiveKey="first" style={{ display: 'flex' }}>
                <Row className="settingsRow">
                    <Col className=" shadowTab justify-content-center uvs-left uvs-right" sm={2}>

                        <Nav variant="pills" className="flex-column marginTop">
                            <Nav.Item className="selectedSetting ">
                                <Nav.Link eventKey="first">User Profile Settings</Nav.Link>
                            </Nav.Item>
                            <Nav.Item className="unselectedSetting">
                                <Nav.Link eventKey="second">Organization Information</Nav.Link>
                            </Nav.Item>
                        </Nav>
                    </Col>

                    <Col className="dropShadow justify-content-center uvs-left uvs-right" sm={8}>
                        <Container className="settingsContentPaneContainer">
                            <Tab.Content>
                                <Tab.Pane eventKey="first">
                                    <Form >
                                        <Form.Group className="usernameInput" controlId="formPlaintextEmail">
                                            <Form.Text className="usernameText">UserName</Form.Text>
                                            <Form.Control
                                                defaultValue={session.username}
                                                onChange={(e) => {
                                                    setUsername(e.target.value);
                                                }}
                                            ></Form.Control>
                                        </Form.Group>
                                        <Form.Group className="emailInput" controlId="formPlaintextEmail">
                                            <Form.Text className="emailText">Email</Form.Text>
                                            <Form.Control disabled defaultValue={session.email}
                                                onChange={(e) => {
                                                    setEmail(e.target.value);
                                                }}></Form.Control>
                                        </Form.Group>
                                    </Form>
                                    <Button 
                                        className="settingsSaveButton"
                                        onClick={() => SaveUpdatedUserInfo()}>
                                        Save
                                    </Button>
                                </Tab.Pane>
                                <Tab.Pane eventKey="second">
                                </Tab.Pane>
                            </Tab.Content>
                        </Container>
                    </Col>
                </Row>
            </Tab.Container>
        </>
    );
}
export default SettingsMenu;

