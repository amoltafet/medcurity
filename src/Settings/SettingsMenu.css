.profileImage {
    width: 5vmax;
    height: 5vmax;
}

.settingsRow {
    width: 90vmax;
}

.nav-link.active {
    width: 8vmax;
    border-radius: 5px;
    background-color: #cc3333 !important;
    text-align: center;
}

.unselectedSetting {
    color: #1e5b88;
    text-align: center;
    border: #1e5b88;
    width: 8vmax;
    border-radius: 5px;  
}

.unselectedSetting:hover {
    color: #d8dae3;
}

.settingsMenuContainer {
    margin-top: 1vmax;
    margin-bottom: 2vmax;
    margin-left: 2vmax;
    margin-right: 2vmax;
    height: 20vmax;
}

.usernameText { 
    font-family: Lato,sans-serif;
    font-size: 1.4vmax;
    margin-top: 2vmax;
    margin-bottom: 2vmax;
}

.usernameControl {
    border: #d8dae3;
}


.emailText {
    font-family: Lato,sans-serif;
    font-size: 1.4vmax;
    margin-top: 2vmax;
    margin-bottom: 2vmax;
}

.emailControl {
    border: #d8dae3;
}


.emailInput {
    font-family: Lato,sans-serif;
    font-size: 1.4vmax;
    margin-top: 2vmax;
    margin-bottom: 2vmax;
    width: 60vmax;
}

.usernameInput {
    font-family: Lato,sans-serif;
    font-size: 1.4vmax;
    margin-top: 2vmax;
    margin-bottom: 2vmax;
    width: 60vmax;
}

.dropShadow {
    margin-top: 2vmax;
    margin-bottom: 2vmax;
    margin-left: 2vmax;
    margin-right: 2vmax;
    width: 70vmax;
}

.shadowTab {
    margin-top: 2vmax;
    margin-bottom: 2vmax;
    margin-left: 2vmax;
    margin-right: 2vmax;
    width: 11vmax;
    height: 13vmax;
 
}

.marginTop {
    margin-top: 1vmax;
}
