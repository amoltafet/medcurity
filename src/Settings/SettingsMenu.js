import { Nav, Row, Form, Tab, Col, Container, Button, Popover, OverlayTrigger, Image } from 'react-bootstrap';
import { useEffect, useState } from "react";
import { useNavigate } from 'react-router-dom';
import axios from 'axios';
import React from 'react';
import './SettingsMenu.css';
import 'bootstrap/dist/css/bootstrap.min.css';

/**
* Creates and displays the Settings menu allows the user to toggle between diffrent settings. 
* @return {GetPage}
*/
const SettingsMenu = () => {
    axios.defaults.withCredentials = true;
    const [currentUser, setCurrentUser] = useState([]);
    const [newUserName, setUsername] = useState("");
    const [saveData, setSaveData] = useState(false);
    const [show, setShow] = useState(false);
    const [isLoaded, setLoaded] = useState(false);
    const [company, setCompany] = useState([]);
    const [dueDate, setDueDate] = useState([]);
    const [newPassword, setPassword] = useState("");
    const [repeatPassword, setRepeatPassword] = useState("");
    const [convertedProfilePhoto, setConvertedProfilePicture] = useState("/user.png");
    const [profilePic, setProfilePic] = useState("")
    const [message, setMessage] = useState("Saved!")
    const navigate = useNavigate();


    useEffect(() => { if (currentUser.userid) axios.get("http://localhost:3002/api/getProfilePicture", { params: { id: currentUser.userid }} ).then((response) => { setProfilePic(response.data.profileImage) }); })

    useEffect(() => {
        axios.get("http://localhost:3002/users/login").then((response) => {
            setCurrentUser(response.data.user[0]);
        }).catch(error => console.error(`Error ${error}`));
        setLoaded(true)
    }, []);

    useEffect(() => {
        if (isLoaded) {
            axios.get('http://localhost:3002/api/getQuery', 
                { params: { the_query: 'SELECT * FROM AffiliatedUsers WHERE UserID = ' + currentUser.userid} 
            }).then((response) => {
                if (response.data[0].datejoined !== null) { 
                    setCompany(response.data[0]);
                    var date = new Date(response.data[0].datejoined);
                    setDueDate(date.toDateString()); 
                
                }
            }).catch(error => console.error(`Error ${error}`));
        }
    },[currentUser, dueDate, isLoaded])

 

    useEffect(() => {
        if (saveData === true) 
        {
            if (newUserName !== "") {
                axios.post("http://localhost:3002/users/changeUserName", {
                    username: newUserName,
                    id: currentUser.userid
                }).then((response) => {
                    console.log("response", response.data);
                    
                }).catch(error => console.log(`Error ${error}`));
            }
            if(newPassword !== "") {
                axios.post("http://localhost:3002/users/changeUserPassword", {
                    userid: currentUser.userid,
                    newPassword: newPassword,
                    retypedPassword: repeatPassword
                }).then((response) => {
                    console.log("response", response.data);
                    setMessage(response.data["message"])
                    console.log("Message:", message)
                    
                }).catch(error => console.log(`Error ${error}`));
            }

            setSaveData(false);
            setPassword("")
            setRepeatPassword("")
            setShow(true);

        }
       
    }, [saveData, newUserName, currentUser.userid])

    setTimeout(() =>{
        setShow(false)
    }, 9000)

   

    function uploadUserPhoto (userPhoto) {
        /*console.log("unconverted photo", userPhoto)
        var convertedPhoto = URL.createObjectURL(userPhoto);
        console.log("convertedPhoto: ", convertedPhoto)
        setConvertedProfilePicture(convertedPhoto);*/

        //TODO ... THEN call API method to store the image from (banner)
        var data = new FormData();
        data.append("profileImage", userPhoto);
        axios.post("http://localhost:3002/api/postProfilePicture", data, { params: { userid: currentUser.userid } , headers: { 'Content-Type': 'multipart/form-data' } })

    }
  
    
    function SaveUpdatedUserInfo() {
        setSaveData(true);
    }

    const NavToDash = () => {
        navigate('/dash');
      };

    const popover = (
        <Popover id="popover-basic">
            <Popover.Title as="h3">{message}</Popover.Title>
        </Popover>
    );


    return (
        <>
            <Tab.Container className="justify-content-center" defaultActiveKey="first" style={{ display: 'flex' }}>
                <Col className="settingsRow uvs-left uvs-right">
                    <Row className="dropShadow justify-content-center">
                        <Nav variant="pills selectionBox" className="selectionBox justify-content-center">
                            <Nav.Item className="justify-content-center settingSpacing">
                                <Nav.Link className=' justify-content-center selectedSetting' eventKey="first">Profile Settings</Nav.Link>
                            </Nav.Item>
                            <Nav.Item className="justify-content-center settingSpacing">
                                <Nav.Link className=' unselectedSetting justify-content-center' eventKey="second">Company Info</Nav.Link>
                            </Nav.Item>
                        </Nav>
                    </Row>
                    <Row className="dropShadow justify-content-center">
                        <Container className="settingsContentPaneContainer">
                        <Tab.Content>
                        <Tab.Pane eventKey="first">
                            <Form>
                                <Form.Group className="justify-content-center">
                                    <Image className="settingsProfilePicture" variant="top" src={`data:image/png;base64,${profilePic}`} alt="" roundedCircle></Image>
                                <Form.File 
                                    className="userProfilePhotoInput"
                                    onChange={(e) => uploadUserPhoto((e.target.files[0]))}
                                    accept=".png,.jpg,.jpeg"/>
                                </Form.Group>
                                <Form.Group className="emailInput" controlId="formPlaintextEmail">
                                        <Form.Text className="emailText">Email</Form.Text>
                                        <Form.Control disabled defaultValue={currentUser.email}></Form.Control>
                                </Form.Group>
                                <Form.Group className="usernameInput" controlId="formPlaintextEmail">
                                    <Form.Text className="usernameText">Username</Form.Text>
                                    <Form.Control
                                        defaultValue={currentUser.username}
                                        onChange={(e) => {
                                            setUsername(e.target.value);
                                        }}>
                                    </Form.Control>
                                </Form.Group>
                                <h3>Change Password</h3>
                                <Form.Group className="passwordInput" controlId="formPlaintextEmail">
                                    <Form.Text className="passwordText" id="newPasswordText">New Password</Form.Text>
                                    <Form.Control
                                        type="password"
                                        value = {newPassword}
                                        onChange={(e) => {
                                            setPassword(e.target.value);
                                        }}
                                    ></Form.Control>
                                </Form.Group>
                                <Form.Group className="passwordInput" controlId="formPlaintextEmail">
                                    <Form.Text className="passwordText" id="repeatPasswordText">Retype Password</Form.Text>
                                    <Form.Control
                                        type="password"
                                        value = {repeatPassword}
                                        onChange={(e) => {
                                            setRepeatPassword(e.target.value);
                                        }}
                                    ></Form.Control>
                                </Form.Group>
                            </Form>
                               <OverlayTrigger delay={{ show: 5000, hide: 4000 }} show={show} placement="bottom" overlay={popover}>
                                        <Button
                                            className="settingsWideButton"
                                            onClick={() => SaveUpdatedUserInfo()}>
                                            Save
                                        </Button>
                                    </OverlayTrigger>
                                        <Button
                                            className="settingsWideButton"
                                            onClick={NavToDash}>
                                            Back to Dashboard
                                        </Button>
                                   
                                </Tab.Pane>
                                <Tab.Pane eventKey="second">
                                    <Form>
                                    <Form.Group className="usernameInput" controlId="formPlaintextEmail">
                                        <Form.Text className="usernameText">Company</Form.Text>
                                        <Form.Control disabled defaultValue={company.name}></Form.Control>
                                    </Form.Group>
                                    <Form.Group className="emailInput" controlId="formPlaintextEmail">
                                            <Form.Text className="emailText">Date Joined</Form.Text>
                                            <Form.Control disabled defaultValue={dueDate}></Form.Control>
                                    </Form.Group>
                                    </Form>
                                </Tab.Pane>
                            </Tab.Content>
                        </Container>
                    </Row>
                </Col>
            </Tab.Container>
        </>
    );
}
export default SettingsMenu;

