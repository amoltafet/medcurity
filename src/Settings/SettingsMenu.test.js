import renderer from 'react-test-renderer';
import SettingsMenu from './SettingsMenu';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

test('renders SettingsMenu', () => {
  const tree = renderer.create(<BrowserRouter><SettingsMenu /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});