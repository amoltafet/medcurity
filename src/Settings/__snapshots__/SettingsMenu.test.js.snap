// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginPage 1`] = `
<div
  className="settingsRow row"
>
  <div
    className="shadowTab col-sm-2"
  >
    <div
      className="flex-column marginTop  nav nav-pills"
      onKeyDown={[Function]}
      role="tablist"
    >
      <div
        className="selectedSetting nav-item"
      >
        <a
          aria-controls="left-tabs-example-tabpane-first"
          aria-selected={true}
          className="nav-link active"
          data-rb-event-key="first"
          href="#"
          id="left-tabs-example-tab-first"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="tab"
        >
          <img
            alt=""
            className="profileImage  rounded-circle"
            src="/user.png"
            variant="top"
          />
        </a>
      </div>
      <div
        className="unselectedSetting nav-item"
      >
        <a
          aria-controls="left-tabs-example-tabpane-second"
          aria-selected={false}
          className="nav-link"
          data-rb-event-key="second"
          href="#"
          id="left-tabs-example-tab-second"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="tab"
        >
          Tab 2
        </a>
      </div>
    </div>
  </div>
  <div
    className="dropShadow col-sm-8"
  >
    <div
      className="settingsMenuContainer container"
    >
      <div
        className="tab-content"
      >
        <div
          aria-hidden={false}
          aria-labelledby="left-tabs-example-tab-first"
          className="fade tab-pane active show"
          id="left-tabs-example-tabpane-first"
          role="tabpanel"
        >
          <form
            className=""
          >
            <div
              className="usernameInput form-group"
            >
              <small
                className="usernameText form-text"
              >
                UserName
              </small>
              <input
                className="form-control"
                id="exampleForm.ControlTextarea1"
                placeholder="username"
              />
            </div>
            <div
              className="emailInput form-group"
            >
              <small
                className="emailText form-text"
              >
                Email
              </small>
              <input
                className="form-control"
                id="exampleForm.ControlTextarea1"
                placeholder="email"
              />
            </div>
          </form>
        </div>
        <div
          aria-hidden={true}
          aria-labelledby="left-tabs-example-tab-second"
          className="fade tab-pane"
          id="left-tabs-example-tabpane-second"
          role="tabpanel"
        />
      </div>
    </div>
  </div>
</div>
`;
